cli/src/main/scala/com/advancedtelematic/tuf/cli/repo/TufRepo.scala

```diff
@@ -131,6 +165,64 @@ class TufRepo(val name: RepoName, val repoPath: Path)(implicit ec: ExecutionCont
     }
   }
 
+  private def fetchRootChain(reposerverClient: UserReposerverClient, from: Int, to: Int): Future[Vector[SignedPayload[RootRole]]] = {
+    val versionsToFetch = from until to
+
+    val rootRolesF = versionsToFetch.foldLeft(Future.successful(Vector.empty[SignedPayload[RootRole]])) { (accF, vv) =>
+      for {
+        acc <- accF
+        payload <- reposerverClient.root(vv.some).recoverWith {
+          case RoleNotFound(msg) => Future.failed(RootPullError(NonEmptyList.of(msg)))
+        }
+      } yield acc :+ payload
+    }
+
+    rootRolesF
+  }
+
+  private def validatePath(reposerverClient: UserReposerverClient, from: SignedPayload[RootRole], to: SignedPayload[RootRole]): Future[Unit] = {
+    val rootRolesF = fetchRootChain(reposerverClient, from.signed.version + 1, to.signed.version)
+
+    rootRolesF.flatMap { rootRoles =>
+      val validated = (from +: rootRoles :+ to).sliding(2, 1).toList.traverse {
+        case Vector(oldR, newR) =>
+          RootRoleValidation.newRootIsValid(newR, oldR)
+        case l =>
+          throw new IllegalArgumentException(s"too many elements for $l")
+      }
+
+      validated.fold(
+        errors => Future.failed(RootPullError(errors)),
+        _ => Future.successful(())
+      )
+    }
+  }
+
+  def pullRoot(reposerverClient: UserReposerverClient, skipLocalValidation: Boolean): Future[SignedPayload[RootRole]] = {
```

Comment:
The idea with offline signing is that there should never be reliance on serer side validation. If someone could compromise the server to push a bad 1.root.json, a dev would sign that and then push a bad 2.root.json (or whatever). Since the sever is pwnd, it would push that bad metadata out to devices. I guess documentation explaining the danger would be helpful.